lib/plugins/plugin/install/install.js

```diff
@@ -109,6 +109,14 @@ class PluginInstall {
 
   addPluginToServerlessFile() {
     return this.getServerlessFilePath().then(serverlessFilePath => {
+      if (_.last(_.split(serverlessFilePath, '.')) === 'js') {
+        this.serverless.cli.log(`
+          Can't automatically add plugin into "serverless.js" file.
```

Comment:
Why cannot a plugin be installed automatically?

Reply:
Because the js file only exports the serverless object, but hides the details how that is kept or generated. So the plugin plugin does not have any chance to tell the serverless.js to include the new plugin into the exported object the next time it is loaded.
In theory the whole service definition could be dynamically compiled depending on some environment settings. That way we have to treat the serverless.js as opaque and the exposed object as read-only.

Reply:
OK, I see :smile: